File: src/main/java/com/rs/java/game/area/banks/CamelotBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.shapes.Polygon;

public final class CamelotBank extends Area {

	public CamelotBank() {
		super(
				new Polygon(new WorldTile[] {
						new WorldTile(2724, 3487, 0),
						new WorldTile(2724, 3490, 0),
						new WorldTile(2721, 3490, 0),
						new WorldTile(2721, 3494, 0),
						new WorldTile(2719, 3494, 0),
						new WorldTile(2719, 3496, 0),
						new WorldTile(2721, 3496, 0),
						new WorldTile(2721, 3497, 0),
						new WorldTile(2730, 3497, 0),
						new WorldTile(2730, 3490, 0),
						new WorldTile(2727, 3490, 0),
						new WorldTile(2727, 3487, 0),
						new WorldTile(2724, 3487, 0) // closing point
				})
		);
	}

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Camelot Bank";
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}
}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/CamelotBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class CamelotBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2724, 3487, 0),
            WorldTile(2724, 3490, 0),
            WorldTile(2721, 3490, 0),
            WorldTile(2721, 3494, 0),
            WorldTile(2719, 3494, 0),
            WorldTile(2719, 3496, 0),
            WorldTile(2721, 3496, 0),
            WorldTile(2721, 3497, 0),
            WorldTile(2730, 3497, 0),
            WorldTile(2730, 3490, 0),
            WorldTile(2727, 3490, 0),
            WorldTile(2727, 3487, 0),
            WorldTile(2724, 3487, 0) // closing point
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Camelot Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
